cachito/workers/pkg_managers/pip.py

```diff
@@ -1448,6 +1477,115 @@ def _sdist_preference(sdist_pkg):
     return yanked_pref, filetype_pref
 
 
+def _download_vcs_package(requirement, pip_deps_dir, pip_raw_repo_name, nexus_auth):
+    """
+    Fetch the source for a Python package from VCS (only git is supported).
+
+    After downloading, upload this package to the Nexus raw repository and on subsequent
+    downloads, reuse the uploaded asset instead of fetching from VCS again.
+
+    :param PipRequirement requirement: VCS requirement from a requirements.txt file
+    :param Path pip_deps_dir: The deps/pip directory in a Cachito request bundle
+    :param str pip_raw_repo_name: Name of the Nexus raw repository for Pip
+    :param requests.auth.AuthBase nexus_auth: Authorization for the Nexus raw repo
+
+    :return: Path to downloaded file
+    """
+    git_info = _extract_git_info(requirement.url)
+
+    namespace_parts = git_info["namespace"].split("/")
+    repo_name = git_info["repo"]
+    ref = git_info["ref"]
+
+    # Download to e.g. deps/pip/github.com/namespace/repo
+    package_dir = pip_deps_dir.joinpath(git_info["host"], *namespace_parts, repo_name)
+    package_dir.mkdir(parents=True, exist_ok=True)
+
+    filename = f"{repo_name}-external-gitcommit-{ref}.tar.gz"
+    download_path = package_dir / filename
+
+    # Check if we already have the raw component
+    raw_component_name = f"{repo_name}/{filename}"
+    log.debug("Looking for raw component %r in %r repo", raw_component_name, pip_raw_repo_name)
+    download_url = nexus.get_raw_component_asset_url(pip_raw_repo_name, raw_component_name)
```

Comment:
Note for future self: this gets the component in the nexus hoster instance, if available.